server/swap/swap.go

```diff
@@ -1132,145 +1159,172 @@ func (s *Swapper) makerRedeemStatus(mStatus *swapStatus, tAsset uint32) (makerRe
+	if s.orders.swapSuccess(otherOrder) {
+		// TODO: We should count this as a successful swap, but should
+		// it only be a completed order with the extra stipulation that
+		// it had already completed another swap?
+		compTime := time.Now().UTC()
+		s.authMgr.RecordCompletedOrder(otherOrder.User(), otherOrder.ID(), compTime)
+		if err := s.storage.SetOrderCompleteTime(otherOrder, encode.UnixMilli(compTime)); err != nil {
+			if db.IsErrGeneralFailure(err) {
+				log.Errorf("fatal error with SetOrderCompleteTime for order %v: %v", otherOrder.UID(), err)
+			} else {
+				log.Warnf("SetOrderCompleteTime for %v: %v", otherOrder.UID(), err)
+			}
+		}
+	}
+
+	// Penalize for failure to act.
+	//
+	// TODO: Arguably, this obviates the RecordCancel above since this
+	// closes the account before the possibility of a cancellation ratio
+	// penalty. I'm keeping it this way for now however since penalties
+	// may become less severe than account closure (e.g. temporary
+	// suspension, cool down, or order throttling), and restored
+	// accounts will still require a record of the revoked order.
+	s.authMgr.Penalize(orderAtFault.User(), account.FailureToAct)
+
+	// Send the revoke_match messages, and solicit acks.
+	s.revoke(match)
+}
+
+func (s *Swapper) checkInactionEventBased() {
 	// If the DB is failing, do not penalize or attempt to start revocations.
 	if err := s.storage.LastErr(); err != nil {
 		log.Errorf("DB in failing state.")
 		return
 	}
 
 	var deletions []*matchTracker
-	oldestAllowed := time.Now().Add(-s.bTimeout).UTC()
+
+	// Do time.Since(event) with the same now time for each match.
+	now := time.Now()
+	tooOld := func(evt time.Time) bool {
+		return now.Sub(evt) >= s.bTimeout
+	}
 
 	checkMatch := func(match *matchTracker) {
 		// Lock entire matchTracker so the following is atomic with respect to
 		// Status.
 		match.mtx.RLock()
 		defer match.mtx.RUnlock()
 
-		if match.makerStatus.swapAsset != assetID && match.takerStatus.swapAsset != assetID {
-			return
-		}
+		log.Tracef("checkInactionEventBased() => checkMatch(%v, %v)",
+			match.ID(), match.Status)
 
 		failMatch := func(makerFault bool) {
-			orderAtFault, otherOrder := match.Taker, order.Order(match.Maker) // an order.Order
-			if makerFault {
-				orderAtFault, otherOrder = match.Maker, match.Taker
+			s.failMatch(match, makerFault)
+			deletions = append(deletions, match)
+		}
+
+		switch match.Status {
+		case order.NewlyMatched:
+			// Maker has not broadcast their swap. They have until match time
+			// plus bTimeout.
+			if tooOld(match.time) {
+				failMatch(true) // maker should have swapped
+			}
+		case order.MakerRedeemed:
+			// If the maker has redeemed, the taker can redeem immediately, so
+			// check the timeout against the time the Swapper received the
+			// maker's `redeem` request (and sent the taker's 'redemption').
+			match.makerStatus.mtx.RLock()
+			defer match.makerStatus.mtx.RUnlock()
+			if tooOld(match.makerStatus.redeemTime) {
```

Comment:
Server sends the redemption to taker a brief moment after setting the redeemTime; and taker may not receive the server-sent redemption until `auth.DefaultRequestTimeout + auth.DefaultConnectTimeout` later. It'd be basically pointless resending the request if taker reconnects after `s.bTimeout` because the match would have been revoked by then.

Reply:
Same comment regarding the brief moment.  I don't think we need to account for a few milliseconds.

I do agree it is pointless to send the request after bTimeout, so ideally the connect timeout would be capped at bTimeout.